django/core/checks/settings_deprecation.py

```diff
@@ -0,0 +1,86 @@
+from django.conf import settings
+from django.core import checks
+
+
+def deprecated_settings_variables(*args, **kwargs):
+    """
+    This check warns users who still use deprecated settings variables.
+    """
+
+    deprecated_settings_list = {
+        # Django 1.2
+        "DATABASE_ENGINE",
+        "DATABASE_HOST",
+        "DATABASE_NAME",
+        "DATABASE_OPTIONS",
+        "DATABASE_PASSWORD",
+        "DATABASE_PORT",
+        "DATABASE_USER",
+        "TEST_DATABASE_CHARSET",
+        "TEST_DATABASE_COLLATION",
+        "TEST_DATABASE_NAME",
+        # Django 1.4
+        "TRANSACTIONS_MANAGED",
+        # Django 1.5
+        "AUTH_PROFILE_MODULE",
+        # Django 1.7
+        "SOUTH_DATABASE_ADAPTER",
+        "SOUTH_DATABASE_ADAPTERS",
+        "SOUTH_AUTO_FREEZE_APP",
+        "SOUTH_TESTS_MIGRATE",
+        "SOUTH_LOGGING_ON",
+        "SOUTH_LOGGING_FILE",
+        "SOUTH_MIGRATION_MODULES",
+        "SOUTH_USE_PYC",
+        # Django 1.8
+        "SEND_BROKEN_LINK_EMAILS",
+        "CACHE_MIDDLEWARE_ANONYMOUS_ONLY",
+        # Django 1.10
+        "ALLOWED_INCLUDE_ROOTS",
+        "LOGOUT_URL",
+        "TEMPLATE_CONTEXT_PROCESSORS",
+        "TEMPLATE_DEBUG",
+        "TEMPLATE_DIRS",
+        "TEMPLATE_LOADERS",
+        "TEMPLATE_STRING_IF_INVALID",
+        # Django 2.0
+        "MIDDLEWARE_CLASSES",
+        # Django 2.1
+        "USE_ETAGS",
+        "SECURE_BROWSER_XSS_FILTER",
+        # Django 3.0
+        "DEFAULT_CONTENT_TYPE",
+        "PASSWORD_RESET_TIMEOUT_DAYS",
+        # Django 3.1
+        "DEFAULT_FILE_STORAGE",
+        "FILE_CHARSET",
+        # Django 4.0
+        "DEFAULT_HASHING_ALGORITHM",
+        "PASSWORD_RESET_TIMEOUT_DAYS",
+        "SECURE_BROWSER_XSS_FILTER",
+        # Django 5.0
+        "USE_L10N",
+        "USE_DEPRECATED_PYTZ",
+        "CSRF_COOKIE_MASKED",
+        "DATABASE_OPTIONS",
+        # todo: DATABASES->name->TEST->SERIALIZE not yet covered
```

Comment:
@matthiask What do you think about these nested settings? a) should we check them and b) would you go for a separate approach or make the set more verbose?

Reply:
What would the failure mode of `SERIALIZE` be? If there's a potential for data loss as well I'd probably try adding a check for it too. If not, I'm not so sure. 

My instinct is to add this later, in a separate PR/ticket anyway, because there will probably be many more cases for things like this, and reaching an agreement could be harder. I'd try getting the simpler change merged first, and look at deeper inspection later.

Reply:
> My instinct is to add this later, in a separate PR/ticket anyway, because there will probably be many more cases for things like this, and reaching an agreement could be harder. I'd try getting the simpler change merged first, and look at deeper inspection later.

Exactly my thoughts ❤️ Didn't want to push it without some concent, though. I'm removing the todo.